dimscord/objects.nim

```diff
@@ -577,74 +618,55 @@ proc newEmoji*(data: JsonNode): Emoji =
         for r in data["roles"].elems:
             result.roles.add(r.str)
 
-    if data.haskey("id") and data["id"].kind != JNull:
-        result.id = data["id"].str
-
-    if data.hasKey("user"):
-        result.user = newUser(data["user"])
-    if data.hasKey("require_colons"):
-        result.require_colons = data["require_colons"].bval
-    if data.hasKey("managed"):
-        result.managed = data["managed"].bval
-    if data.hasKey("animated"):
-        result.animated = data["animated"].bval
+    result.assignFrom(data):
+        id <<- it.str
+        user <- newUser(it)
+        require_colons <- it.bval
+        managed <- it.bval
+        animated <- it.bval
 
 proc newGameActivity*(data: JsonNode): GameActivity =
     result = GameActivity(
         name: data["name"].str,
         kind: data["type"].getInt(),
         created_at: data["created_at"].num
     )
-    if data.hasKey("url") and data["url"].kind != JNull:
-        result.url = some(data["url"].str)
     
     if data.hasKey("timestamps"):
-        result.timestamps = some((start: toBiggestInt(0), final: toBiggestInt(0)))
-        if data["timestamps"].hasKey("start"):
-            get(result.timestamps).start = data["timestamps"]["start"].num
-        if data["timestamps"].hasKey("end"):
-            get(result.timestamps).final = data["timestamps"]["end"].num
+        result.timestamps = some((
+            start: data["timestamps"]{"start"}.getInt.BiggestInt,
+            final: data["timestamps"]{"end"}.getInt.BiggestInt
+        ))
     
-    if data.hasKey("application_id"):
-        result.application_id = some(data["application_id"].str)
-    if data.hasKey("details") and data["details"].kind != JNull:
-        result.details = some(data["details"].str)
-    if data.hasKey("state") and data["state"].kind != JNull:
-        result.state = some(data["state"].str)
-    if data.hasKey("emoji"):
-        result.emoji = some(newEmoji(data["emoji"]))
+    result.assignFrom(data):
+        url <<- some(it.str)
+        application_id <- some(it.str)
+        details <<- some(it.str)
+        state <<- some(it.str)
+        emoji <- some(newEmoji(it))
+        instance <- it.bval
+        flags <- it.getInt()
+
     if data.hasKey("party"):
-        result.party = some((id: "", size: ""))
-        if data["party"].hasKey("id"):
-            get(result.party).id = data["party"]["id"].str
-        if data["party"].hasKey("size"):
-            # TODO: Parse MaxSize too?
-            get(result.party).size = $data["party"]["size"][0].getInt()
+        result.party = some((
+            id: data["party"]{"id"}.getStr,
+            size: data["party"]{"size"}.getStr

```

Comment:
It seems that the `getInt` line change occurred on your `devel` branch. This needs to be reflected in my changes.

Reply:
Although the type of `size` is still `string`, as defined in objects.nim:174?
```nim
party*: Option[tuple[id: string, size: string]]
```